File: 8-string-to-integer-atoi/8-string-to-integer-atoi.c
int myAtoi(char * s){
    long int res = 0;
    int sign = 1;
    int i = 0;
    
    while(s[i] == 32)   //in ASCII 32: ' '(space)
        i++;
    
    if(s[i] == '-'){
        sign *= -1;
        i++;
    }
    else if(s[i] == '+')
        i++;
    
    while(s[i] && (s[i] >= '0' && s[i] <= '9')){
        res = (res * 10) + s[i] - '0';
        if(res*sign > INT_MAX)    //overflow
            return INT_MAX;
        if(res*sign < INT_MIN)    //underflow
            return INT_MIN;
        i++;
    }
    return res * sign;
}

File: 8-string-to-integer-atoi/8-string-to-integer-atoi.java
class Solution {
    public int myAtoi(String s) {
        long res = 0;
        int sign = 1;
        int i = 0;
        
        if(s.length() == 0 || s == null)
            return 0;
        
        while(i < s.length() && s.charAt(i) <= 32)
            i++;
        
        if(i < s.length() && (s.charAt(i) == '+' || s.charAt(i) == '-')){
            if(s.charAt(i) == '-')
                sign *= -1;
            i++;
        }
        
        while(i < s.length() && s.charAt(i) >= '0' && s.charAt(i) <= '9'){
            res = res * 10 + (s.charAt(i) - '0');
            i++;
            if(res * sign > Integer.MAX_VALUE)
                return Integer.MAX_VALUE;
            else if(res * sign < Integer.MIN_VALUE)
                return Integer.MIN_VALUE;
        }
        
        return (sign * (int)res);
    }
}